### ISSUE TITLE
Netflix isn't working even with Widevine correctly loaded.

### ISSUE THREAD

**Issue by yoannmoinet** [electron/electron#12427]
Hi there, I have an issue when opening Netflix with Electron.
I had a Widevine plugin working for some time, and it started to fail unexpectedly.

Could you help troubleshoot this please?

* Electron version: 1.8.4
* Operating system: macOS High Sierra 10.13.3

* My environment 

| env | version |
|:----|-----:|
| ares | "1.10.1-DEV" |
| atom-shell | "1.8.4" |
| chrome | "59.0.3071.115" |
| electron | "1.8.4" |
| http_parser | "2.7.0" |
| modules | "57" |
| node | "8.2.1" |
| openssl | "1.0.2l" |
| uv | "1.13.1" |
| v8 | "5.9.211.38" |
| zlib | "1.2.11" |

* Plugins

<img width="738" alt="screen shot 2018-03-24 at 11 16 43" src="https://user-images.githubusercontent.com/597828/37862892-e12343ae-2f54-11e8-9393-7df6a4bb5334.png">

### Expected behavior

When widevine plugin is correctly loaded, it should play Netflix.

### Actual behavior

Getting an error in Netflix, eventhough shaka and bitmovin show a correctly loaded and activated widevine plugin.

<img width="677" alt="screen shot 2018-03-24 at 10 21 28" src="https://user-images.githubusercontent.com/597828/37862874-9df41f54-2f54-11e8-8781-483d285ec6a3.png">

### How to reproduce

I've created a very simple repo which will help reproduce the issue easily.
You'll be able to also load Bitmovin and Shaka to ensure Widevine is indeed correctly enabled.

```shell
git clone git@github.com:yoannmoinet/netflix-and-electron.git
cd netflix-and-electron
npm install
npm start
```

Thank you so much for your help.

**Comment by sofianguy** [comment]
Thanks for reaching out!

Because we treat our issues list as the team's backlog, we close issues that are questions since they don't represent a task needing to be completed. For most questions about Electron there are a lot of options. 
Check out the [Electron community](https://github.com/electron/electron#community). There are also a bunch of helpful people in this [community forum](https://discuss.atom.io/c/electron) that should be willing to point you in the right direction.

**Comment by yoannmoinet** [comment]
It might have been formuled as a question, but I do think it's a bug on Electron's part.

And I did everything needed to describe it as well as I could.

**Comment by yoannmoinet** [comment]
Even though this issue has been closed, I've new informations for any new person that will come here for answers, because this is an Electron issue nonetheless.

It appears that Netflix implemented a new VMP (verified media path) support to the Widevine browser CDM lately. So the little players/browsers won't be able to play DRM content without a proper, whitelisted, Widevine CDM installation coming from Google.

Brave got the same issue, Vivaldi as well.

I've contacted them to see what are the steps to get a legit widevine plugin.

Basically, it means that Electron based apps won't be able to play Netflix anymore.

All Hail DRM.

**Comment by levrik** [comment]
@yoannmoinet Really? I just started building an Netflix Desktop app based on Electron...

**Comment by yoannmoinet** [comment]
@levrik I confirm, there is no solution as of now. I'm still waiting after Google to get a widevine legit plugin. But I'm not really confident.

In the mean time, I'm working on a shady solution. We'll see where that takes me.

If you get it to work, I'm really interested, because I have a desktop app, published on the AppStore that isn't working with Netflix anymore.

**Comment by levrik** [comment]
@yoannmoinet What is the app you have published?

**Comment by yoannmoinet** [comment]
https://fenêt.re

**Comment by levrik** [comment]
@yoannmoinet uh.. an paid app. That's bad that it isn't working anymore. In the meanwhile I found this: https://github.com/castlabs/electron-releases
It gets it's Widevine binaries from this URL https://redirector.gvt1.com/edgedl/widevine-cdm/1010-win-x64.zip which seems to be Google owned according to whois.

**Comment by yoannmoinet** [comment]
Thank you @levrik, I've tried your link. 
Unfortunately it's missing the adapter needed by Electron.

I might try using the Brave's engine [Muon](https://github.com/brave/muon) that is a fork of Electron, they have a [different way to register the widevine plugins](https://github.com/brave/browser-laptop/blob/master/app/extensions.js#L611).

**Comment by levrik** [comment]
@yoannmoinet The adapter is inside the GitHub releases of the linked repo. Also wanted to try out Muon. Will do so tomorrow after work.

**Comment by levrik** [comment]
@yoannmoinet Do you plan do write an message to that mail address from the `LICENSE.txt`? If yes, would be interested in the out-come.

**Comment by yoannmoinet** [comment]
I might, I've already contacted Google on another email address (widevine-support@google.com) to get a legit plugin. I'll try again on this one.

**Comment by levrik** [comment]
@yoannmoinet Nice, please keep me posted. Also via mail if you don't want to post that public (see my GitHub profile for that).

**Comment by ccj242** [comment]
@yoannmoinet any updates on this? I have contacted widevine myself and have yet to hear back.

**Comment by yoannmoinet** [comment]
@ccj242 unfortunately still nothing from Google (widevine).
I'll keep you posted here once I have something new.

**Comment by alexanderturinske** [comment]
Looking at the `castlabs/electron-releases` repo, it seems several have been able to build using it and the certificates they received from Google; does this mean that the Widevine pluggin is no longer viable in versions of electron >v1.6.18?

**Comment by alexanderturinske** [comment]
@yoannmoinet Did they get back to you?

**Comment by ccj242** [comment]
To share my experience, my non-profit sent 3 certified letters to widevine headquarters in addition to countless online forms and emails sent and have received no word whatsoever. Very frustrating. I'm beginning to worry this is a lost cause.

**Comment by alexanderturinske** [comment]
My company uses [DRMToday](https://castlabs.com/drmtoday/) (paid-service owned by [castLabs](https://castlabs.com/open-source/downstream/)) and they were able to get us the necessary items, though it still took months.

**Comment by yoannmoinet** [comment]
@alexanderturinske I've contacted widevine-support@google.com and they answered me 1.5 month later.

They asked if I had completed the 'Widevine MLA'. 
I had no idea what it was, so I asked them.

No news for more than 2 weeks now. I'll ping them again.

**Comment by yoannmoinet** [comment]
I've passed the one year anniversary waiting for my license.
Still no license, but I've finally signed an agreement a few weeks ago.

Waiting for the next steps.

**Comment by ccj242** [comment]
Don't get your hopes up, I signed my agreement a few months ago and after repeated emails, still have no response.

**Comment by a-sync** [comment]
related: https://blog.samuelmaddock.com/posts/google-widevine-blocked-my-browser/

**Comment by yoannmoinet** [comment]
4 months after signing the agreement, still no license. 😞

**Comment by ccj242** [comment]
Same boat.... totally unbelievable.

**Comment by ccj242** [comment]
I can confirm nothing on my end either, 6 months and counting since I signed the agreement.

**Comment by yoannmoinet** [comment]
My app might be getting featured on the AppStore in the coming weeks, so I'll add a message when visiting Netflix, Hulu or PrimeVideo. 
Explaining to the user that Google still hasn't given me a license to stream those protected content.

Maybe, I could add some link to post on twitter and call Google out on their retention.

**Comment by alexanderturinske** [comment]
I believe the process will be greatly expedited (if not necessary at this point) for a developer/company to become/go through a [Certified Widevine Implementatin Patner](https://www.widevine.com/training/#partners) in order to receive the necessary certificates. That link has a list of partners that seem to provide widevine support and I would try reaching out to one of them for assistance. As previously mentioned, we went through [castLabs](https://castlabs.com/solutions/drm/) and it still took months.

**Comment by fightwithcrayon** [comment]
@ccj242 @yoannmoinet To tie this story off, did either of you ever get anything in the end?

I may also need to go the Implementation Partner route – does anyone know the cost typically involved in this option?

**Comment by ccj242** [comment]
So funny you should ask this now -- just two weeks ago Widevine sent me the certificates totally out of the blue! This after they sent me the agreement a year ago, - finally closing this two year saga. I had all but given up and I have no idea why they suddenly decided to do it. There was no charge, but perhaps it was because my organization is a non-profit? Anyway, fairytale endings occasionally come true after two years of waiting? At long last here is my newly updated widevine-working electron app: https://www.cinqmarsmedia.com/chameleonvideoplayer/

**Comment by AlexanderAllen** [comment]
Just in case it helps anyone, here's a writeup of implementing widevine in Electron over at Medium. 

- Using stock Electron: https://medium.com/@callback.insanity/spotify-web-playback-sdk-feat-electron-50ff930d5c74
- Using castLabs Electron: https://medium.com/@callback.insanity/castlabs-widevine-cdm-ea369bb5623b
- Github repo for both stories: https://github.com/callbackinsanity/Et-in-Electron-Ego-III

Big shoutout to @yoannmoinet and @ccj242 for persisting through two years of insanity!! This issue was my fountain of wisdom and hope when all else failed.

**Comment by samuelmaddock** [comment]
> Using castLabs Electron: https://medium.com/@callback.insanity/castlabs-widevine-cdm-ea369bb5623b

By default the castLabs fork uses a version of Widevine signed for development purposes. Unless you complete a license agreement with Widevine, many websites—including Netflix and Hulu—will fail to load. [A VMP-signed production binary](https://github.com/castlabs/electron-releases#verified-media-path-vmp) is required for these websites.
